[14.0][ADD] sale_barcode: add module

dessanhemrayev opened the pull request:
This module adds the product to the sales order line according to the barcode information provided by the barcode/datamatrix device.

dessanhemrayev committed:
[ADD] sale_barcode: add module

dessanhemrayev committed:
[DON'T MERGE] test-requirements.txt

dessanhemrayev commented:
Create lot and/or select lot if used in combination with sale_order_lot_selection feature is not implemented yet

aayartsev commented on sale_barcode/models/sale_order.py:
```diff
@@ -0,0 +1,20 @@
+    @api.model
+    def _product_barcode_search(self, barcode):
+        """Returns the found product by barcode"""
+        return self.env["product.product"].search([("barcode", "=", barcode)], limit=1)
```
Do we need to write limit=1 when we have same constraint in model?

dessanhemrayev replied:
Thanks) I'll fix it

dessanhemrayev committed:
[IMP] sale_barcode: Increase qty in SOL

- scanning twice the same product should increase qty instead of adding new line

aayartsev reviewed (approved):
It looks good
